types/database/DatabaseClient.js

```diff
@@ -120,13 +125,13 @@ export default class DatabaseClient {
   }
 
   fetchDBInfo(userID) {
-    return new Promise(async (resolve, reject) => {
-      await this.createDBInfo(userID)
-      await this.database.collection('users').findOne({
-          userID
-        })
-        .then(resolve)
-        .catch(reject)
+    return this.createDBInfo(userID)
+  }
+
+  updateXP(userID, xp) {
```

Comment:
The person using this method would only be putting in the amount of XP to increment by, rather than the final XP to set. We should fetch the current XP and calculate level, then we should increment (`$inc`) the XP by the specified amount instead of overwriting it.